mixer/adapter/prometheus/server_test.go

```diff
@@ -32,6 +32,16 @@ func TestServer(t *testing.T) {
 		t.Fatalf("Start() failed unexpectedly: %v", err)
 	}
 
+	s3 := newServer(testAddr)
+
+	if err := s3.Start(test.NewEnv(t), http.HandlerFunc(doesNothing)); err != nil {
```

Comment:
this can't possibly work, right?  in the lines above you started a server on the exact same port.

Checking the presubmit jobs, I see:

```
server_test.go:38: Start() failed unexpectedly: could not start prometheus metrics server: listen tcp 127.0.0.1:9992: bind: address already in use
```

Reply:
Also, does `s3` need to be closed?  Or should these lines be testing for failure?

Reply:
You are right, I was trying out something just before sending the PR. It does not work